providers/amazon/src/airflow/providers/amazon/aws/sensors/eks.py

```diff
@@ -145,6 +145,7 @@ def __init__(
         **kwargs,
     ):
         super().__init__(target_state=target_state, target_state_type=ClusterStates, **kwargs)
+        self.region = region
```

Comment:
Is this the right fix? I think we should replace the template field to region_name in a backward compatible way

Reply:
We need to save it as an attribute because it is listed in template fields. Any field in template fields should also be an attribute.

Reply:
Correct me if I am wrong but...
If region_name is templated and we assign the vakue to it, then we can just remive region from the list without causing a breaking change

Reply:
That would not break dags if a user set `region` using a template?

Reply:
Since we save the value in `region_name` and it is the templated field, maybe not actually. As far as I know, the interpretation of templated fields happens between the init and the execute? If yes, then yes I think you're right

Reply:
I updated the PR, I think now it makes more sense.